src/main/java/io/zonky/test/db/postgres/junit5/PreparedDbExtension.java

```diff
@@ -55,13 +54,33 @@ public void beforeAll(ExtensionContext extensionContext) throws Exception {
         provider = PreparedDbProvider.forPreparer(preparer, builderCustomizers);
         connectionInfo = provider.createNewDatabase();
         dataSource = provider.createDataSourceFromConnectionInfo(connectionInfo);
+        perClass = true;
```

Comment:
This should only get set when the extension is static due to junit 5 only calling `@beforeAll` and `@afterAll` for static extensions.